core/tee/tee_svc_storage.c

```diff
@@ -598,7 +663,15 @@ TEE_Result tee_svc_storage_obj_create(uint32_t storage_id, void *object_id,
 		goto exit;
 	}
 
+clean:
+	tee_file_ops.unlink(initfile);
+
 exit:
+	free(file);
+	file = NULL;
+	free(initfile);
+	initfile = NULL;
+
 	if (res != TEE_SUCCESS) {
 		if (fd >= 0)
 			tee_file_ops.close(fd);
```

Comment:
The code is rather hard to read. Here, `fd` corresponds to the renamed file. But `po` had been created on the temporary file `.xxx`. So I am not sure we should call `tee_pobj_release(po);` in case the error has been raised after the rename. 

Any way to restructure the code?